lib/Driver/Driver.cpp

```diff
@@ -1653,6 +1653,15 @@ Driver::computeCompilerMode(const DerivedArgList &Args,
   if (!ArgRequiringSingleCompile)
     return OutputInfo::Mode::StandardCompile;
 
+  // AST dump doesn't work in WMO. Since it's not common want to dump the AST,
+  // we assume that's the priority and ignore `-wmo`, but we warn the user about
+  // this decision.
+  if (Args.hasArg(options::OPT_whole_module_optimization) &&
```

Comment:
You're ignoring OPT_index_file here, compared to the logic above. If that's intentional, it deserves a comment.

Reply:
Oh never mind, adding `-index-file` to the check seems to have worked after all.

Reply:
I would just reuse the boolean from above, so that it's always consistent.

Reply:
Ok, I'm reusing the boolean, but I think a separate if statement makes the code easier to understand... do you agree?